### PR TITLE
Traefik

### PR DESCRIPTION
**Pull request by johnpyp**

### CODE REVIEW

File: server/docker-compose.yml [hunk 1]
```diff
@@ -40,23 +32,19 @@ services:
       - ${REDIS_PORT}
 
   traefik:
-    image: traefik:1.7
+    image: traefik:latest
```
Comment (alg1142) [inline server/docker-compose.yml]: never use latest

File: server/docker-compose.yml [hunk 2]
```diff
@@ -40,23 +32,19 @@ services:
...
-      - 'traefik.enable=true'
-      - 'traefik.http.routers.traefik1_router_http.rule=Host(`traefik.${DOMAIN}`)'
-      - 'traefik.http.middlewares.https_redirect.redirectscheme.scheme=https'
-      - 'traefik.http.routers.traefik1_router_http.middlewares=https_redirect'
-      - 'traefik.http.routers.traefik1_router_https.rule=Host(`traefik.${DOMAIN}`)'
-      - 'traefik.http.routers.traefik1_router_https.tls'
-      - 'traefik.http.services.traefik1_service.loadbalancer.server.port=8080'
+      traefik.enable: 'true'
+      traefik.backend: 'traefik'
+      traefik.port: '8080'
```
Comment (alg1142) [inline server/docker-compose.yml]: if you're adding this add a user & password into the env file and put it behind auth

File: server/docker-compose.yml [hunk 3]
```diff
@@ -11,18 +11,10 @@ services:
...
-      traefik.backend: rustlesearch_api
-      traefik.docker.network: traefik
-      traefik.port: ${APP_PORT}
-      traefik.web.frontend.rule: Host:${HOSTNAME}
-      traefik.web.frontend.headers.browserXSSFilter: true
-      traefik.web.frontend.headers.contentTypeNosniff: true
-      traefik.web.frontend.headers.forceSTSHeader: true
-      traefik.web.frontend.headers.STSPreload: true
-      traefik.web.frontend.headers.frameDeny: true
-      traefik.web.frontend.headers.STSSeconds: 315360000
```
Comment (alg1142) [inline server/docker-compose.yml]: readd the headers, however its done on 1.7

File: server/.env.sample [hunk 4]
```diff
@@ -2,7 +2,6 @@ LOG_LEVEL=debug
 APP_PORT=3000
 NODE_ENV=test
 ELASTIC_LOCATION=http://localhost:9200
-HOSTNAME=api.rustlesearch.dev
```
Comment (alg1142) [inline server/.env.sample]: readd this, name it API_HOSTNAME, add equivalent in config check and in log message

File: server/config/traefik/traefik.toml [hunk 5]
```diff
@@ -5,32 +5,26 @@ logLevel = "WARN"
...
+   ]
+   minVersion = "VersionTLS12"
 
 [acme]
 entryPoint = "web"
 storage = "/etc/traefik/acme/acme.json"
-onHostRule = false
-resolvers = ["1.1.1.1:53", "1.0.0.1:53"]
-caServer = "https://acme-v02.api.letsencrypt.org/directory"
-KeyType = "EC384"
```
Comment (alg1142) [inline server/config/traefik/traefik.toml]: readd these

File: server/.env.sample (hunk 4 above)
Comment (alg1142) [reply server/.env.sample]: removed entirely in https://github.com/johnpyp/rustlesearch-server/pull/28/commits/c804fbfdc456e4906f6faa6dd37e9d085dee5fd5

**Comment by johnpyp** [comment]
nginx sucks

**Pull request closed without merging**